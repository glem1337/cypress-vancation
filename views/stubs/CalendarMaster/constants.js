import moment from 'moment';

export const currentDateRange = {
  from: moment('2021-04-12').startOf('day'),
  to: moment('2021-04-25').endOf('day'),
};

export const currentDate = moment('2021-04-19');

export const VANS_STATUSES = {
  published: 'Published',
  unpublished: 'Unpublished',
};

export const vansMock = [
  {
    info: {
      title: 'Adventure Ready Class B Camper Rocky Mountains',
      slug: 'super-van-145645',
      status: VANS_STATUSES.published,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-16 14:00'),
        to: moment('2021-04-19 12:00'),
        prices: [400, 500, 550],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-21 12:00'),
        prices: [400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-10 12:00'),
        to: moment('2021-04-14 14:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class B Camper Rocky Mountains',
      slug: 'super-van-145646',
      status: VANS_STATUSES.published,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-12 14:00'),
        to: moment('2021-04-13 12:00'),
        prices: [400],
      },
      {
        from: moment('2021-04-14 14:00'),
        to: moment('2021-04-16 12:00'),
        prices: [400, 450],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-24 12:00'),
        prices: [400, 450, 400, 400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-13 14:00'),
        to: moment('2021-04-14 20:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class 2',
      slug: 'super-van-145647',
      status: VANS_STATUSES.published,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-16 14:00'),
        to: moment('2021-04-19 12:00'),
        prices: [400, 500, 550],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-21 12:00'),
        prices: [400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-17 14:00'),
        to: moment('2021-04-20 12:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class 2',
      slug: 'super-van-145648',
      status: VANS_STATUSES.unpublished,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-12 14:00'),
        to: moment('2021-04-13 12:00'),
        prices: [400],
      },
      {
        from: moment('2021-04-14 14:00'),
        to: moment('2021-04-16 12:00'),
        prices: [400, 450],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-24 12:00'),
        prices: [400, 450, 400, 400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-11 16:00'),
        to: moment('2021-04-28 12:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
  {
    info: {
      title: 'Adventure Ready Class 2',
      slug: 'super-van-145649',
      status: VANS_STATUSES.unpublished,
      image: {
        imgUrl: 'https://placeimg.com/68/68/nature',
        imgUrlRetina: 'https://placeimg.com/136/136/nature',
      },
    },

    dates: [
      {
        from: moment('2021-04-16 14:00'),
        to: moment('2021-04-19 12:00'),
        prices: [400, 500, 550],
      },
      {
        from: moment('2021-04-20 14:00'),
        to: moment('2021-04-21 12:00'),
        prices: [400],
      },
    ],

    bookings: [
      {
        from: moment('2021-04-23 16:00'),
        to: moment('2021-04-27 12:00'),
        details: {
          peopleCount: 2,
          nightsCount: 2,
          total: 1500,
        },
      },
    ],
  },
];
