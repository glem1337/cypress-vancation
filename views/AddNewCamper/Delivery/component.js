import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Col, Row, Form, Skeleton } from 'antd';

import { TYPE_DELIVERY } from 'constants/mapbox';
import ProgressBar from 'views/shared/ProgressBar';
import AddNewCamperBtnForm from 'views/shared/AddNewCamperBtnForm';
import Map from 'views/shared/Delivery/Map';
import Pickup from 'views/shared/Delivery/Pickup';
import Rate from 'views/shared/Delivery/Rate';

import EstimatedEarningCard from '../EstimatedEarningCard';

const Delivery = ({
  onBackButtonClick,
  handlerPickup,
  handlerRate,
  isLoading,
  handleSubmit,
  isFormValid,
  values,
  camper,
  camperCompleteness,
  isCamperExist,
}) => {
  if (!isCamperExist) {
    return (
      <Row>
        <Col span={24} className="d-flex justify-content-center">
          <div className="main-listing-container">
            <div className="mb-16 mb-md-24">
              <ProgressBar
                content={<FormattedMessage id="addNewCamper.ProgressBar" />}
                percent={camperCompleteness}
              />
            </div>
            <Row gutter={24}>
              <Col lg={24}>
                <h1 className="text-headline mb-8">
                  <FormattedMessage id="addNewCamper.delivery.willYouOfferDelivery" />
                </h1>
              </Col>
              <Col lg={24}>
                <Skeleton />
              </Col>
            </Row>
          </div>
        </Col>
      </Row>
    );
  }

  return (
    <div className="main-listing-delivery__wrapper">
      <Row className="d-block">
        <Col span={24} className="d-flex justify-content-center">
          <div className="main-listing-container main-listing-delivery__container">
            <div className="mb-16 mb-md-24">
              <ProgressBar
                content={<FormattedMessage id="addNewCamper.ProgressBar" />}
                percent={camperCompleteness}
              />
            </div>
            <Form layout="vertical">
              <Row gutter={24}>
                <Col lg={24}>
                  <h1 className="text-headline mb-8">
                    <FormattedMessage id="addNewCamper.delivery.willYouOfferDelivery" />
                  </h1>
                  <p className="mb-24">
                    <FormattedMessage id="addNewCamper.delivery.ifYouOfferDelivery" />
                  </p>
                </Col>
                <Pickup
                  isLoading={isLoading}
                  rate={values.rate}
                  pickup={values.pickup}
                  handlerRate={handlerRate}
                  handlerPickup={handlerPickup}
                />
                <Col lg={24}>
                  <Rate
                    isLoading={isLoading}
                    hide={!values.pickup || values.rate === TYPE_DELIVERY[0]}
                  />
                </Col>
              </Row>
            </Form>
            <AddNewCamperBtnForm
              withBackBtn
              onBackClick={onBackButtonClick}
              onSaveClick={handleSubmit}
              isSubmitting={isLoading}
              canSave={isFormValid && !isLoading}
              showGradientButton={isFormValid}
            />
          </div>
        </Col>
      </Row>
      <Map
        hide={!values.pickup}
        camper={camper}
        radius={values.distance}
      />
      <EstimatedEarningCard isSlim />
    </div>
  );
};

Delivery.defaultProps = {
  isLoading: false,
  camper: null,
};

Delivery.propTypes = {
  isLoading: PropTypes.bool,
  isFormValid: PropTypes.bool.isRequired,
  onBackButtonClick: PropTypes.func.isRequired,
  values: PropTypes.shape({
    pickup: PropTypes.bool.isRequired,
    rate: PropTypes.oneOf(TYPE_DELIVERY).isRequired,
    distance: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]).isRequired,
  }).isRequired,
  camper: PropTypes.shape({
    specificationDetail: PropTypes.shape(),
  }),
  handlerPickup: PropTypes.func.isRequired,
  handlerRate: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  camperCompleteness: PropTypes.number.isRequired,
  isCamperExist: PropTypes.bool.isRequired,
};

export default Delivery;
