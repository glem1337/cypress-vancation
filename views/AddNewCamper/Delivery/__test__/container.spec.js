import React from 'react';
import { shallow } from 'enzyme';
import * as R from 'ramda';
import ROUTES from 'constants/routes';
import configureStore from 'redux-mock-store';

import { TYPE_DELIVERY, DELIVERY_DEFAULT_VALUES } from 'constants/mapbox';
import diveTo from 'utils/testHelpers/diveToEnzyme';
import {
  checkPreviousStepCompleteness,
  createCamperDeliveryAction,
  updateCamperDeliveryAction,
  fetchCamper,
} from 'state/concepts/camper/actions';
import {
  showModal,
  hideModal,
} from 'state/modal/actions';
import { createRouteFromPathname } from 'utils/createRouteHelper';
import redirect from 'utils/redirect';
import mockedCamper from 'views/__mocks__/camper';

import Delivery, { DeliveryContainer } from '../container';

jest.mock('utils/redirect', () => jest.fn());

jest.mock('utils/form/isSubmitDisabled', () => jest.fn(() => true));

jest.mock('state/data/selectors', () => ({
  loadingSelector: jest.fn(() => true),
}));

jest.mock('state/concepts/camper/selectors', () => ({
  camperSelector: jest.fn(() => mockedCamper),
  camperCompletenessSelector: jest.fn(() => 22),
  isCamperExistSelector: jest.fn(() => true),
  leavePageMethodSelector: jest.fn(() => jest.fn()),
}));

jest.mock('state/data/selectors', () => ({
  loadingSelector: jest.fn(() => false),
}));

jest.mock('formik', () => ({
  withFormik: (config) => (Component) => props => {
    const extended = {
      ...config,
      ...props,
    };

    return <Component {...extended} />;
  },
}));

describe('Map Container', () => {
  let container;
  let wrapper;
  let instance;

  const store = configureStore()({});
  store.dispatch = jest.fn();
  store.logicMiddleware = {
    whenComplete: jest.fn(),
  };

  const props = {
    camperId: mockedCamper.id,
    store,
    router: {
      push: jest.fn(),
    },
    setValues: jest.fn(),
    setSubmitting: jest.fn(),
    setStatus: jest.fn(),
    setErrors: jest.fn(),
    resetForm: jest.fn(),
    createCamperDelivery: jest.fn(),
    updateCamperDelivery: jest.fn(),
    values: DELIVERY_DEFAULT_VALUES,
    validateForm: jest.fn(),
    dirty: true,
    isValid: true,
    setFieldValue: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Delivery {...props} />);
    container = diveTo(wrapper, DeliveryContainer);
    instance = container.instance();

    jest.clearAllMocks();
  });

  it('tests "getInitialProps" static method', async () => {
    const ctx = {
      query: {
        camper_id: 'test id',
      },
      store,
    };

    await DeliveryContainer.getInitialProps(ctx);

    expect(ctx.store.dispatch).toHaveBeenNthCalledWith(
      1,
      fetchCamper(ctx.query.camper_id, 'delivery_information'),
    );

    expect(ctx.store.dispatch).toHaveBeenNthCalledWith(
      2,
      checkPreviousStepCompleteness({
        key: ROUTES.ADD_NEW_CAMPER.DELIVERY.KEY,
        camperId: ctx.query.camper_id,
        ctx,
      }),
    );

    expect(store.logicMiddleware.whenComplete).toHaveBeenCalled();
  });

  it('snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('checks mapPropsToValues configs', () => {
    expect(container.props().mapPropsToValues((container.props()))).toMatchSnapshot();
  });

  it('checks validationSchema configs', () => {
    expect(container.props().validationSchema).toMatchSnapshot();
  });

  it('tests "submitData" instance getter', () => {
    const { camperId } = container.props();

    const result = R.compose(
      R.append({ camperId }),
      R.props(['values', 'setErrors', 'setSubmitting', 'resetForm', 'setErrors', 'setValues']),
    )(container.props());

    expect(instance.submitData).toEqual(result);
  });

  it('tests "isLoading" instance getter', () => {
    expect(instance.isLoading).toEqual(false);
  });

  describe('tests "handlerPickup" instance method', () => {
    it('value = false', () => {
      instance.handlerPickup(false);

      const values = {
        pickup: false,
        rate: DELIVERY_DEFAULT_VALUES.rate,
        distance: DELIVERY_DEFAULT_VALUES.distance,
        costPerMile: DELIVERY_DEFAULT_VALUES.costPerMile,
        minFee: DELIVERY_DEFAULT_VALUES.minFee,
      };

      expect(props.setValues).toHaveBeenCalledTimes(1);
      expect(props.setValues).toHaveBeenCalledWith({
        ...props.values,
        ...values,
      });
    });

    it('value = true', () => {
      instance.handlerPickup(true);
      const values = { pickup: true };

      expect(props.setValues).toHaveBeenCalledTimes(1);
      expect(props.setValues).toHaveBeenCalledWith({
        ...props.values,
        ...values,
      });
    });
  });

  describe('tests "handlerRate" instance method', () => {
    it('value = rates', () => {
      const e = { target: { value: TYPE_DELIVERY[1] } };
      const values = { rate: e.target.value };

      instance.handlerRate(e);

      expect(props.setValues).toHaveBeenCalledTimes(1);
      expect(props.setValues).toHaveBeenCalledWith({
        ...props.values,
        ...values,
      });
    });

    it('value = free', () => {
      const e = { target: { value: TYPE_DELIVERY[0] } };
      const values = {
        rate: e.target.value,
        costPerMile: DELIVERY_DEFAULT_VALUES.costPerMile,
        minFee: DELIVERY_DEFAULT_VALUES.minFee,
      };

      instance.handlerRate(e);

      expect(props.setValues).toHaveBeenCalledTimes(1);
      expect(props.setValues).toHaveBeenCalledWith({
        ...props.values,
        ...values,
      });
    });
  });

  describe('tests "handleSubmit" instance method', () => {
    it('deliveryInformation is not empty', () => {
      instance.handleSubmit();

      const { camperId } = container.props();

      const result = R.compose(
        R.append({ camperId }),
        R.props(['values', 'setErrors', 'setSubmitting', 'resetForm', 'setErrors', 'setValues']),
      )(container.props());

      expect(store.dispatch).toHaveBeenCalledTimes(1);
      expect(store.dispatch).toHaveBeenCalledWith(updateCamperDeliveryAction(...result));
    });

    it('deliveryInformation is empty', () => {
      container.setProps({
        camper: {
          ...mockedCamper,
          deliveryInformation: null,
        },
      });

      instance.handleSubmit();

      const result = R.compose(
        R.append({ camperId: props.camperId }),
        R.props(['values', 'setErrors', 'setSubmitting', 'resetForm', 'setErrors', 'setValues']),
      )(container.props());

      expect(store.dispatch).toHaveBeenCalledTimes(1);
      expect(store.dispatch).toHaveBeenCalledWith(createCamperDeliveryAction(...result));
    });

    it('when is not valid', () => {
      container.setProps({
        isValid: false,
      });

      const res = instance.handleSubmit();

      expect(res).toBe(false);
    });
  });

  describe('tests `onBackButtonClick` instance method', () => {
    it('when form is valid', async () => {
      await instance.onBackButtonClick();

      expect(props.validateForm).toHaveBeenCalled();

      expect(store.dispatch).toHaveBeenCalledWith(
        showModal({
          modalType: 'LEAVE_PAGE_MODAL',
          modalProps: {
            discard: instance.goBack,
            save: instance.saveAndGoBack,
          },
        }),
      );
    });

    it('when form is not valid', async () => {
      container.setProps({
        isValid: false,
      });

      const goBackSpy = jest.spyOn(instance, 'goBack').mockImplementationOnce(() => jest.fn());

      await instance.onBackButtonClick();

      expect(props.validateForm).toHaveBeenCalled();
      expect(store.dispatch).not.toHaveBeenCalled();
      expect(goBackSpy).toHaveBeenCalled();
    });
  });

  it('tests `goBack` instance method', async () => {
    instance.goBack();

    expect(props.router.push).toHaveBeenCalledWith(
      createRouteFromPathname(ROUTES.ADD_NEW_CAMPER.LISTING_PHOTOS.PATH, props.camperId),
    );

    expect(store.dispatch).toHaveBeenCalledWith(
      hideModal(),
    );
  });

  it('tests `saveAndGoBack` instance method', async () => {
    const handleSubmitSpy = jest.spyOn(instance, 'handleSubmit').mockImplementationOnce(() => jest.fn());

    await instance.saveAndGoBack();

    const route = createRouteFromPathname(
      ROUTES.ADD_NEW_CAMPER.LISTING_PHOTOS.PATH,
      props.camperId,
    );

    expect(store.dispatch).toHaveBeenCalledWith(hideModal());
    expect(props.setFieldValue).toHaveBeenCalledWith('redirectRoute', route);
    expect(handleSubmitSpy).toHaveBeenCalled();
  });

  describe('tests `leavePage` instance method', () => {
    it('with saving', async () => {
      const handleSubmitSpy = jest.spyOn(instance, 'handleSubmit');

      await instance.leavePage(true, 'test')();

      expect(props.setFieldValue).toHaveBeenCalledWith('redirectRoute', 'test');
      expect(handleSubmitSpy).toHaveBeenCalled();
      expect(store.dispatch).toHaveBeenCalledWith(hideModal());
    });

    it('without saving', async () => {
      const handleSubmitSpy = jest.spyOn(instance, 'handleSubmit');

      await instance.leavePage(false, 'test')();

      expect(props.setFieldValue).not.toHaveBeenCalled();
      expect(handleSubmitSpy).not.toHaveBeenCalled();
      expect(redirect).toHaveBeenCalledWith('test');
      expect(store.dispatch).toHaveBeenCalledWith(hideModal());
    });
  });
});
