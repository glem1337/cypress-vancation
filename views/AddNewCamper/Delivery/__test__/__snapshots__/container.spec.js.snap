// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Container checks mapPropsToValues configs 1`] = `
Object {
  "costPerMile": 1,
  "distance": 1,
  "minFee": 1,
  "pickup": true,
  "rate": "free",
}
`;

exports[`Map Container checks validationSchema configs 1`] = `
ObjectSchema {
  "_blacklist": ReferenceSet {
    "list": Set {},
    "refs": Map {},
  },
  "_blacklistError": undefined,
  "_excludedEdges": Array [],
  "_nodes": Array [
    "minFee",
    "costPerMile",
    "distance",
  ],
  "_sortErrors": [Function],
  "_typeError": [Function],
  "_whitelist": ReferenceSet {
    "list": Set {},
    "refs": Map {},
  },
  "_whitelistError": undefined,
  "conditions": Array [],
  "deps": Array [],
  "exclusiveTests": Object {},
  "fields": Object {
    "costPerMile": NumberSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "max": true,
        "min": true,
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "number",
    },
    "distance": NumberSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "max": true,
        "min": true,
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "number",
    },
    "minFee": NumberSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "max": true,
        "min": true,
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "number",
    },
  },
  "spec": Object {
    "abortEarly": true,
    "nullable": false,
    "presence": "optional",
    "recursive": true,
    "strict": false,
    "strip": false,
  },
  "tests": Array [],
  "transforms": Array [
    [Function],
  ],
  "type": "object",
}
`;

exports[`Map Container snapshot 1`] = `
<Delivery
  camper={
    Object {
      "blockedPeriods": Array [],
      "camperAddition": Object {
        "amenityHealthSafetyItems": Array [
          Object {
            "healthSafety": Object {
              "iconUrl": "path/to/icon/url",
              "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
              "name": "name",
            },
          },
        ],
        "camperQuestions": Array [
          Object {
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "required": true,
            "text": "test",
          },
        ],
        "customRestrictionRoads": Array [
          Object {
            "active": true,
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "name": "name",
          },
        ],
        "customRestrictionRules": Array [
          Object {
            "active": true,
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "name": "name",
          },
        ],
        "customTravelAccessories": Array [
          Object {
            "active": true,
            "description": "",
            "id": "dd477886-ec22-4da9-ba4c-08627f35fb1a",
            "maxAmount": "2.0",
            "name": "Bike Rack",
            "price": "1.0",
            "priceUnit": "each",
          },
        ],
        "customTravelRestrictions": Array [
          Object {
            "active": true,
            "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
            "name": "name",
          },
        ],
        "restrictionRoad": Object {
          "dirtryRoad": true,
          "fourWheelRoad": false,
          "offRoad": false,
          "snowAndIceRoad": false,
        },
        "restrictionRule": Object {
          "allowPets": false,
          "festivalApproved": false,
          "smoking": false,
        },
        "travelRestriction": Object {
          "burningMan": false,
          "canada": true,
          "mexico": false,
        },
      },
      "camperCalendar": Object {
        "blockedPeriods": Array [
          Object {
            "endDate": "2021-08-06",
            "id": "56ceb535-322f-42c6-8dcd-a88ad59000f3",
            "startDate": "2021-08-05",
          },
          Object {
            "endDate": "2021-08-25",
            "id": "17a32ec0-a486-4247-910c-6c0c57ba6ec3",
            "startDate": "2021-08-25",
          },
          Object {
            "endDate": "2021-07-28",
            "id": "2c7bbaef-6b00-494f-b005-6a762e90c785",
            "startDate": "2021-07-28",
          },
        ],
        "customDiscountPeriods": Array [
          Object {
            "endDate": "2021-07-28",
            "id": "ffb3f0cc-4513-4fdc-81d0-03587232d9ee",
            "monthlyDiscountPercent": 23,
            "startDate": "2021-07-27",
            "weeklyDiscountPercent": 15,
          },
          Object {
            "endDate": "2021-07-28",
            "id": "a070339d-7058-4769-8e16-814084c6bfe3",
            "monthlyDiscountPercent": 23,
            "startDate": "2021-07-28",
            "weeklyDiscountPercent": 15,
          },
          Object {
            "endDate": "2021-07-28",
            "id": "aba16cba-5873-4273-8172-816b276a2db7",
            "monthlyDiscountPercent": 44,
            "startDate": "2021-07-28",
            "weeklyDiscountPercent": 44,
          },
          Object {
            "endDate": "2021-07-29",
            "id": "586016b9-d351-4ca1-ac8a-373e7e3b8832",
            "monthlyDiscountPercent": 33,
            "startDate": "2021-07-29",
            "weeklyDiscountPercent": 15,
          },
        ],
        "externalCalendars": Array [],
        "id": "e1db81b1-8ad5-4e88-9c99-f5c9020148eb",
        "pricingPeriods": Array [
          Object {
            "endDate": "2021-07-28",
            "id": "8aa92a51-ef00-4dc6-8b84-2820869b9278",
            "price": "33.0",
            "startDate": "2021-07-28",
          },
          Object {
            "endDate": "2021-07-30",
            "id": "248ba38e-e18a-467c-996b-36aa4bd73e85",
            "price": "320.0",
            "startDate": "2021-07-29",
          },
        ],
      },
      "camperPhotos": Array [
        Object {
          "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
          "photoUrl1100": "photoUrl1100",
        },
      ],
      "camperRule": Object {
        "bookingApprovalPolicy": "instant_book",
        "cancellationPolicy": "easy_going",
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
      },
      "deliveryInformation": Object {
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
        "pickup": true,
      },
      "description": "desc",
      "estimatedEarning": 1200,
      "externalCalendars": Array [],
      "glamper": true,
      "id": "a18f7338-c2b3-4cee-be4d-579c0a726c1a",
      "insurance": "pending",
      "insuranceInfo": Object {
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
      },
      "latitude": 40.772,
      "longitude": -73.93030000000002,
      "name": "test",
      "owner": Object {
        "account": Object {
          "createdAt": "2021-07-09T13:39:26.587Z",
          "drivingLicenceVerified": false,
          "email": "kateryna.sessa+owner@rubygarage.org",
          "emailVerified": false,
          "id": "12563773-6e8f-4861-8abd-82a78632c6ac",
          "user": Object {
            "avatarUrl": null,
            "createdAt": "2021-07-09T13:39:26.590Z",
            "firstName": "Kat",
            "id": "77015afb-71e3-4b94-a9cb-69a97dc6b358",
            "lastName": "Kat",
            "updatedAt": "2021-07-09T13:39:26.590Z",
          },
        },
        "avatarUrl": null,
        "averageRating": null,
        "businessTitle": null,
        "campersCount": 25,
        "createdAt": "Member since July 2021",
        "description": "description",
        "email": null,
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
        "idVerified": false,
        "type": "owner",
      },
      "place": "Astoria, New York, New York 11102, United States",
      "placeId": "neighborhood.290416",
      "pricingInfo": Object {
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
      },
      "pricingPeriods": Array [],
      "publicId": "111",
      "specificationDetail": Object {
        "drivetrain": "4x4",
        "fuelType": "electric",
        "insideHeight": "low_top",
        "length": "67.89",
        "mileage": "Under 50,000",
        "modelNaming": "Clair Weimann",
        "name": "Katelyn Lowe",
        "seats": 1,
        "sleeps": 1,
        "stateRegistred": "Arkansas",
        "transmission": "automatice",
        "whoBuiltCamper": "Vasya",
        "year": 1585,
      },
      "status": "draft",
      "tripFee": Object {
        "cleaning": 1,
        "customFees": Array [
          Object {
            "frequency": "per_day",
            "id": "id",
            "name": "name",
            "price": 5,
          },
        ],
        "dumpingFee": 2,
        "flue": 3,
        "id": "74a9c587-7b78-4aa5-ba93-ba3d310be127",
        "lateDropOff": 4,
        "tripFeeGenerator": Object {
          "available": 8,
          "limit": true,
          "overage": 9,
        },
        "tripFeeMileage": Object {
          "available": 6,
          "limit": true,
          "overage": 7,
        },
      },
      "vehicleTypeIconUrl": "test.svg",
      "vehicleTypeName": "Modern Van",
    }
  }
  camperCompleteness={22}
  camperId="a18f7338-c2b3-4cee-be4d-579c0a726c1a"
  createCamperDelivery={[Function]}
  dirty={true}
  handleSubmit={[Function]}
  handlerPickup={[Function]}
  handlerRate={[Function]}
  hideModal={[Function]}
  intl={
    Object {
      "formatMessage": [Function],
    }
  }
  isCamperExist={true}
  isFormValid={true}
  isLoading={false}
  isLoadingCreate={false}
  isLoadingUpdate={false}
  isValid={true}
  mapPropsToValues={[Function]}
  onBackButtonClick={[Function]}
  resetForm={[MockFunction]}
  router={
    Object {
      "push": [MockFunction],
    }
  }
  setErrors={[MockFunction]}
  setFieldValue={[MockFunction]}
  setLeavePageMethod={[Function]}
  setStatus={[MockFunction]}
  setSubmitting={[MockFunction]}
  setValues={[MockFunction]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "logicMiddleware": Object {
        "whenComplete": [MockFunction],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateCamperDelivery={[Function]}
  validateForm={[MockFunction]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_excludedEdges": Array [],
      "_nodes": Array [
        "minFee",
        "costPerMile",
        "distance",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "costPerMile": NumberSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "number",
        },
        "distance": NumberSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "number",
        },
        "minFee": NumberSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "number",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
  values={
    Object {
      "costPerMile": 1,
      "distance": 1,
      "minFee": 1,
      "pickup": false,
      "rate": "free",
    }
  }
/>
`;
